fondant/dataset.py

```diff
@@ -166,82 +167,123 @@ def upload(self, save_path):
         return None
 
 
-class FondantComponent:
-    def __init__(self, type="transform"):
-        # note: Fondant spec always needs to be called like this
-        # and placed in the src directory
+class FondantComponent(ABC):
+    """Abstract base class for a Fondant component"""
+
+    def __init__(self):
         self.spec = FondantComponentSpec.from_file("fondant_component.yaml")
-        self.type = type
 
-    def run(self) -> dd.DataFrame:
-        """
-        Parses input data, executes the transform, and creates the output manifest.
+    @abstractmethod
+    def _get_manifest(self, args: argparse.Namespace) -> Manifest:
+        """Abstract method that returns the dataset manifest"""
 
+    @abstractmethod
+    def _add_and_parse_args(self) -> argparse.Namespace:
+        """Abstract method that adds and parses the component arguments"""
+
+    def get_component_arguments(self) -> t.Mapping[str, Argument]:
+        """
+        Dictionary representation of the input and output arguments of a component
         Returns:
-            Manifest: the output manifest
+            Input and output arguments of the component.
         """
-        # step 1: add and parse arguments
-        args = self._add_and_parse_args()
-        # step 2: create Fondant dataset based on input manifest
-        metadata = json.loads(args.metadata)
-        if self.type == "load":
-            # TODO ideally get rid of args.metadata
-            # by including them in the storage args,
-            # getting run_id based on args.output_manifest_path
-            manifest = Manifest.create(
-                base_path=metadata["base_path"],
-                run_id=metadata["run_id"],
-                component_id=metadata["component_id"],
-            )
-        else:
-            manifest = Manifest.from_file(args.input_manifest_path)
-        dataset = FondantDataset(manifest)
-        # step 3: load or transform data
-        if self.type == "load":
-            df = self.load(args)
-            dataset.add_index(df)
-            dataset.add_subsets(df, self.spec)
-        else:
-            # create dataframe, based on component spec
-            df = dataset.load_data(self.spec)
-            # provide this dataframe to the user
-            df = self.transform(
-                df=df,
-                args=args,
-            )
-            # TODO update index, potentially add new subsets
+        component_arguments = {}
+        kubeflow_component_spec = self.spec.kubeflow_specification
+        component_arguments.update(kubeflow_component_spec.input_arguments)
```

Comment:
just to confirm, the custom `args` of a component are part of the `input_arguments`?

Reply:
yes indeed the kubeflow input arguments are derived from the ComponentSpec of the component which contains the arguments